### PR TITLE
Atomically set slug on Artefact

### PR DESCRIPTION
**Pull request by boffbowsh**
Validation prevents us changing slugs on live artefacts, so we need to
call `set` directly, bypassing validation.

This migration hasn't been run on staging/production yet, and failed in preview.

@heathd / @alext, as this is now bypassing a deliberately added validation, are we actually doing the correct thing here by attempting to change the slug? From what I gather, one or both of you suggested this step in the [planning sheet](https://docs.google.com/spreadsheets/d/1xMgOH3ha7dzoPaHJw9VAu9hlCvAm3ifwbBABbuFNMDQ/edit#gid=0). It makes sense to me, but I want to double check first.

### CODE REVIEW

**Comment by heathd** [comment]
looks good.

However I would consider changing this to be a rake task (or script) rather than a migration. The problem with doing these kind of things in a migration is that you have less control over the timing of when the job runs as it's coupled to the deployment process. 

Using a rake task or script will allow you to deploy the code and then when you're ready run the task.

**Comment by boffbowsh** [comment]
Changed it to a Rake task

File: lib/tasks/detailed_guides.rake [hunk 1]
```diff
@@ -1,16 +1,21 @@
...
       unless detailed_guide.slug.start_with?('guidance/')
         old_slug = detailed_guide.slug
         new_slug = "guidance/#{old_slug}"
+
         begin
-          path = Rails.application.url_arbiter_api.reserve_path("/#{new_slug}", "publishing_app" => "whitehall")
-          detailed_guide.slug = new_slug
-          detailed_guide.save!
-          puts "moved #{old_slug} to #{new_slug}"
+          path = Rails.application.url_arbiter_api.reserve_path("/#{new_slug}",
```
Comment (jamiecobbett) [inline lib/tasks/detailed_guides.rake]: I'm coming to this late, and I know this was already in the migration, but if we're reserving the new path, should we release the old one?

**Comment by heathd** [comment]
looks good as a rake task.

I'll leave you to respond to Jamie's question @boffbowsh

File: lib/tasks/detailed_guides.rake (hunk 1 above)
Comment (boffbowsh) [inline lib/tasks/detailed_guides.rake]: Maybe later as a cleanup task, but there will always* be either a 410 or 301 at the old URL so I don't really think it should be released.

**Pull request merged**